backend/audit/test_schemas.py

```diff
@@ -138,30 +138,33 @@ def test_invalid_uei(self):
             .replace("I", "")
             .replace("o", "")
             .replace("O", "")
+            .upper()  # We convert to uppercase in UEIValidationFormView.
         )
         good_uei = "".join(choice(alpha_omit_oi) for i in range(12))
         idx = randrange(12)
 
+        too_short = "".join(choice(alpha_omit_oi) for i in range(11))
+        too_long = "".join(choice(alpha_omit_oi) for i in range(13))
+        zero_start = f"0{''.join(choice(alpha_omit_oi) for i in range(11))}"
         with_punc = good_uei[:idx] + choice(string.punctuation) + good_uei[idx + 1 :]
-        with_ioIO = good_uei[:idx] + choice("ioIO") + good_uei[idx + 1 :]
-
-        bad_ueis = [
-            "".join(choice(alpha_omit_oi) for i in range(11)),  # too short
-            "".join(choice(alpha_omit_oi) for i in range(13)),  # too long
-            f"0{''.join(choice(alpha_omit_oi) for i in range(11))}",  # starts with 0
-            with_punc,  # contains a non-alphanum char
-            with_ioIO,  # contains one of i, o, I, O
+        with_numlike = good_uei[:idx] + choice("ioIO") + good_uei[idx + 1 :]
+
+        bad_ueis_and_messages = [
+            (too_short, f"'{too_short}' is too short"),
+            (too_long, f"'{too_long}' is too long"),
+            (zero_start, "does not match"),
+            (with_punc, "does not match"),
+            (with_numlike, "does not match"),
```

Comment:
While we're here, and before we merge, it would be good to add some tests that make sure we never have 9 digits in a row. We could do it randomly, or 

1. One check for a UEI that starts with 9 digits
2. One check for a UEI containing 9 digits in the middle
3. One check for a UEI that ends with 9 digits

Those are all invalid, as a UEI should never contain something that could be confused for an EIN or SSN. Hence, the rule about not having 9 digits in a row.

I *think* our regexs catch this. However, I'm not 100% certain. I'd feel better knowing we handled those three cases, as the first and third represent the "edges," so to speak.